lib/lighthouse/benefits_documents/configuration.rb

```diff
@@ -87,7 +87,8 @@ def generate_upload_body(document_data, file_body)
 
       file = Tempfile.new(document_data[:file_name])
       File.write(file, file_body)
-      payload[:file] = Faraday::UploadIO.new(file, 'application/pdf')
+      mime_type = MimeMagic.by_path(document_data[:file_name]).type
```

Comment:
Is there a place where we disallow uploading unsupported file types other than the frontend?

Reply:
I thought about restricting non supported filetypes in the BE but figured for security reasons it needs to be all done on the front end anyway and having redundant logic for it just introduces more points of failure. I could see something slight deviations in file type causing issue (jpg vs jpeg or some other edge cases).

Reply:
If I understand this conversation correctly, @aherzberg  is saying we already check the mime type on the FE so it's best just to rely on that. I'm not sure I agree with that. There are all kinds of places where we do data validations on the backend that are redundant versions of front end checks. Phone numbers, for instance. In general, I think the role of the backend developer is to ensure the safety and reliability of the back end regardless of the front end. In theory, this code should work for the mobile and web apps as well as any other front end that might end up using this code down the road. So while we may trust our current front end, who's to say that's the only front end? Also, keep in mind that support old versions of the mobile front end. Imagine that we need to change the allowed mime types on the front end. What happens to older clients? 

And I think this touches upon my suggestion to have a test for the failure case. I assume Kevin is suggesting we whitelist which mime types are available and raise an error if any disallowed types are uploaded. That would make the failure case testable. 

On a bit of a side note, if the failure case isn't really testable because it just returns a job id, what's the point in testing with both PDF and JPEG? Isn't the type effectively irrelevant in the tests?

Reply:
Continuing on the subject... I suspect below you're saying that we don't really have a way of knowing in the controller if there's a problem because this happens in a job. @kjduensing may have more thoughts into where we should do validations. If we do them in the controller, yes, these tests will not care about the type. But in that case, we could write controller tests for the failure case. It may also be possible to have validation logic in the service object that we also use in the controller, which could allow us to validate both in the controller (which has the advantage of informing the FE of errors) and in the service object if Kevin likes doing validations there. 

I also can somewhat understand your reluctance to add more validations given that we're really just a middle layer and there are validations on the FE and probably upstream as well. I'm not sure we have a systematic approach to that problem but I do know it's no uncommon for us to do validations in our layer.

Reply:
that's fair. I will add the mime type validation.

Reply:
validation is up. let me know if you see any issues.